src/container/include/container/container_parsing.hpp

```diff
@@ -0,0 +1,107 @@
+#ifndef CLIPPER_CONTAINER_PARSING_HPP
+#define CLIPPER_CONTAINER_PARSING_HPP
+
+#include <cstring>
+#include <vector>
+
+#include <container/datatypes.hpp>
+
+namespace clipper {
+
+namespace container {
+
+template <typename T>
+struct input_parser_type {
+  static const bool is_string_parser = false;
+};
+
+template <>
+struct input_parser_type<char> {
+  static const bool is_string_parser = true;
+};
+
+template <typename D>
```

Comment:
Added the following docstring

```
/**
 * Parses raw data of type `D` to create `Input<D>` objects that will be
 * used by models to render predictions.
 *
 * @tparam D A primitive datatype corresponding to a valid Input<D> object
 * (ex: D == `uint8_t` is valid because ByteVector == Input<uint8_t> is a valid Input object)
 */
```